props/saber_fett263_buttons.h

```diff
@@ -355,6 +1485,308 @@ SaberFett263Buttons() : PropBase() {}
+#define STYLE_SETTINGS 10
+#endif
+
+  enum MenuType {
+    MENU_TOP,
+    MENU_PRESET,
+    MENU_VOLUME,
+    TRACK_PLAYER,    
+    MENU_REHEARSE,
+#ifdef FETT263_EDIT_MODE_MENU
+    MENU_STYLE,
+    MENU_COLOR,
+    MENU_COLOR_CHANGE,
+    MENU_FONT,
+    MENU_TRACK,
+    MENU_LENGTH,
+    MENU_COPY,
+    MENU_DELETE,
+    MENU_RESET,
+    MENU_EXIT,
+    MENU_COLOR_SUB,
+    MENU_COLOR_MODE,
+    MENU_EFFECT,
+    MENU_RESET_COLOR,
+#if (NUM_BLADES > 1)
+    MENU_COPY_COLOR,
+    MENU_BLADE_STYLE,
+    MENU_BLADE_COLOR,
+    MENU_BLADE_COPY,
+    MENU_BLADE_SETTING,
+    MENU_BLADE_LENGTH,
+#endif
+    MENU_GESTURE_SUB,
+    MENU_SWINGON,
+    MENU_TWISTON,
+    MENU_THRUSTON,
+    MENU_STABON,
+    MENU_FORCEPUSH,
+    MENU_TWISTOFF,
+    MENU_POWERLOCK,
+    MENU_SETTING_SUB,
+    MENU_STYLE_SETTING_SUB,
+    MENU_STYLE_OPTION,
+    MENU_IGNITION_OPTION,
+    MENU_IGNITION_TIME,
+    MENU_RETRACTION_OPTION,
+    MENU_RETRACTION_TIME,
+    MENU_CLASH_LOCATION,
+    MENU_DRAG_SIZE,
+    MENU_EMITTER_SIZE,
+    MENU_PREON_OPTION,
+    MENU_PREON_SIZE,
+    MENU_DIM_BLADE,
+    MENU_CLASH_THRESHOLD,
+    MENU_SWINGON_SPEED,
+    MENU_FORCEPUSH_LENGTH,
+    MENU_LOCKUP_DELAY,
+    MENU_CLASH_DETECT,
+    MENU_MAX_CLASH,
+    COLOR_BASE,
+    COLOR_ALT,
+    COLOR_BLAST,
+    COLOR_CLASH,
+    COLOR_LOCKUP,
+    COLOR_DRAG,
+    COLOR_LB,
+    COLOR_STAB,
+    COLOR_PREON,
+    COLOR_IGNITE,
+    COLOR_RETRACT,
+    COLOR_PSTOFF,
+    COLOR_SWING,
+    COLOR_EMITTER,
+    COLOR_OFF,
+#endif  
+  };
+
+  enum EditColorMode {
+    NONE,
+    CC_COLOR_LIST,
+#ifdef FETT263_EDIT_MODE_MENU
+    COLOR_LIST,
+    EDIT_BLACK,
+    EDIT_COLOR,
+    EDIT_WHITE,
+    ZOOM_COLOR,
+#endif
+  };
+
+#define CC_NEW_COLOR(N) NewColor(N,1);
+
+// Update Style
+  virtual void UpdateStyle(int preset_num) {
+    TRACE(PROP, "start");
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+#define UNSET_BLADE_STYLE(N) \
+    delete current_config->blade##N->UnSetStyle();
+    ONCEPERBLADE(UNSET_BLADE_STYLE)
+    current_preset_.SetPreset(preset_num);
+#ifdef DYNAMIC_BLADE_LENGTH
+    savestate_.ReadINIFromSaveDir("curstate");
+#define WRAP_BLADE_SHORTERNER(N) \
+    if (savestate_.blade##N##len != -1 && savestate_.blade##N##len != current_config->blade##N->num_leds()) { \
+      tmp = new BladeShortenerWrapper(savestate_.blade##N##len, tmp);	\
+    }
+#else
+#define WRAP_BLADE_SHORTERNER(N)
+#endif
+
+    
+#define SET_BLADE_STYLE(N) do {						\
+    BladeStyle* tmp = style_parser.Parse(current_preset_.current_style##N.get()); \
+    WRAP_BLADE_SHORTERNER(N)                                            \
+    current_config->blade##N->SetStyle(tmp);				\
+  } while (0);
+
+    ONCEPERBLADE(SET_BLADE_STYLE)
+
+#ifdef SAVE_COLOR_CHANGE
+    SaberBase::SetVariation(current_preset_.variation);
+#else
+    SaberBase::SetVariation(0);
+#endif
+    TRACE(PROP, "end");
+  }
+
+// Update Font/ Save Style
```

Comment:
Please expand comment to explain how this is different form SetPreset()

Reply:
comment added, this is simplified to only handle what is necessary in Edit Mode Previews and uses FastOn to bypass Preon effects except if Preon previews are needed in Edit Mode